---
uint16 tool_mask